pysonos/discovery.py

```diff
@@ -156,17 +156,30 @@ def create_socket(interface_addr=None):
                 if b"Sonos" not in data:
                     continue
 
-                # pylint: disable=not-callable
-                zone = config.SOCO_CLASS(addr[0])
-                if zone in seen:
+                # Players can be discovered on multiple interfaces and
+                # we do not want to requery the same player once per interface
+                # if we recently polled it in this discovery
+                if addr[0] in seen:
```

Comment:
I wonder if it is querying all the speakers aka `all_zones` when the instance is created.  I was seeing a flurry of activity in the debug log each discovery.

Reply:
If no interface is specified, we query all. Depending on the network setup that can indeed cause duplicate responses.

However, my point is that your modification should work the same as before because `zone` was already a singleton?

Reply:
I agree it shouldn't be needed since its a singleton.